index.js

```diff
@@ -9,41 +9,96 @@ function defaultInspect() {
     : '[' + this.name + ']'
 }
 
+function defaultToJson() {
```

Comment:
There are a lot more changes here than for #7. It's hard for me to know what the purpose of a lot of these changes are when you did not communicate what issues you're trying to solve. What are you solving with this `toJSON` implementation? What is happening with the default `JSON.stringify` result that makes this necessary?